Support PISC with  pipelines(tna_simple_router, tiny_router_v2 without vlan module)

PFtyrant opened the pull request:
Supporting PISC with (tna_simple_router, tiny_router_v2)
-Fix Makefile
-Fix import path of main.go
-Fix go-bfrt to pisc, bfcli to pisc-cli
-Add set-flow, del-flow, dump, info, table, version features.
-Add pre, port, vlan features.

[README.md](https://github.com/P4Networking/documents/blob/pisc_cli_doc_and_vlan_proto/pisc-cli/README.md)

PFtyrant committed:
Modify info, dump message format

-Modify info, dump message format
-Add all of the table flag in dump.go
-Add PISC logo into version.go
-Fix Makefile
-Fix import path of main.go
-Fix go-bfrt to pisc, bfcli to pisc-cli

PFtyrant committed:
Fix dump bug

-Fix dump error of out of range case

cmingou reviewed (commented):
Can you have more comment about function in utils.go. The format can reference in https://blog.golang.org/godoc

PFtyrant committed:
Add Documents of pisc-cli

- Add README.md
- Add description of util's function
- Remove unused variable
- Fix info that error doesn't occur when table Id is not found

cmingou commented on cmd/utils.go:
```diff
@@ -51,26 +55,115 @@ func initConfigClient() (*p4.BfRuntimeClient, *context.Context, *grpc.ClientConn
-	if ok == true {
-		fmt.Printf("Data Name: %s \n", name)
-		return FOUND
+	//name, ok = p4Info.SearchActionParameterNameById(id)
+	//if ok == true {
+	//	return name, found
+	//}
```
If this code is not used, remove please

PFtyrant replied:
done.

cmingou commented on README.md:
````diff
@@ -0,0 +1,121 @@
+    ```
+2. all : delete all entries of the table
+    ```
+    //Do not confuse with set-flow's "-a" flag. It's totally diffrent.
````
/ 和 D 之間少一個空格

PFtyrant replied:
Fixed it.

cmingou reviewed (changes requested):
LGTM, just need to remove unused code.

PFtyrant committed:
Fix README

-Fix README blank in message
-Remove unused code in utils.go

PFtyrant committed:
Add set/del flow function (#3)

PISC-CLI
-Add set-flow, del-flow, info, table, version function.

* Fixed or removed issues
- Fix table completion to remove the Parser table showing up.
- Fix del-flow out of index problem.
- Remove feature that deleting entry by entry number

* New feature
- Add deleting entry by match keys
- Modify the write request(delete) to delete multiple entities at once.
- Modify the dump function(print format)
- Add go.mod
- Add table name suggestion
- Add dump counts function
- Add the dump count flag -c to take out a number of entries

* Document related
- Modify README for the new version of the flow setup method
- Modify flow cheatsheet

PFtyrant committed:
Fix del-flow function in tune with PISC

PFtyrant committed:
Add read file feature

- Make pisc-cli compatible with tiny_router_v2 pipeline.
- Modify dump command print format. Using tablewriter package.
- Modify info command to use string concatenation instead io buffer.
- Add read file feature to set-flow function

*W.I.P
- Show the progress using progress bar when large number of entries are come.
- make the code more compact

PFtyrant committed:
Fix readfile feature

- Fix the counting error of the readfile feature
- Add progress bar to del-flow command
- Add progress bar to set-flow command(readfile only)
- Fix del-flow bugs
- Tested set-flow/del-flow performance(del-flow has performance issue,
Readtable function of PISC is too slow.)

W.I.P
- Fix suggestion function
- Adding PRE related function(non p4 Object),

PFtyrant committed:
Add table and action suggestion

-Tab completion provides table,action name when use set/del-flow, dump and info command

PFtyrant committed:
Add pre and port feature

-Adding PRE related features(MGID, NODE, LAG, PRUNE)
-Adding Port feature. Port command only support add port config now. It can't modify or delete port config.

PFtyrant committed:
add VLAN features

-Added the VLAN "create" and "add" features
-Fixed the flag shorthand collision problem. Now, the Server flag uses the capital charactor
-Modified PRE command's tab completion to suggest the table name with "pre." suffix
-Modified dump function bug when pipeline table key field counts don't matched to the bfrt profile

PFtyrant committed:
Implement delete feature

-Add delete feature. But not fully implemet. It just deletes one vlan group now.

PFtyrant committed:
modify the vlan add feature to support range config, receive server response message

-Now, the vlan add feature support range config
-read the response message from server

PFtyrant committed:
Add the features: vlan show by port and show by vlan, fix help typo

-Now, the "show" of VLAN cmd is available. vlan show has two options, one is show vlan by vlanId, the other is show vlan by portId.
the usage of the "show" command please refer to the document in P4Networking/doc repo.

PFtyrant committed:
Modify vlan delete and show features.

-use JSON.marshal to decode gRPC response message.
-add show all vlans features. "pisc-cli vlan show" command will show all of the vlans
-add range delete feature.

PFtyrant committed:
Delete the vlan modify feature

-Delete the vlan modify feature.

PFtyrant committed:
Fix set-flow/del-flow bugs

-Modifed the set-flow error messages
-Fixed that the set-flow tab completion suggests abnomaly lists.
-Fixed that the del-flow can't delete the flow when using -m flag.
